### PR TITLE
[eroprofile] Update regexes

### PR DESCRIPTION
**Pull request by YetAnotherGhost**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

It updates the regular expressions following the eroprofile.com site update pointed out in #23200. The title is hopefully stable as long as they stick to `<h1>` elements; the thumbnail is a bit more fragile, but at least it's optional. (I could only find it in one place on the page, in the playlist bar above the video, together with other thumbnails.)

Note that the first video in `_TESTS` appears to no longer exist, but I assume that's unrelated.

### CODE REVIEW

**Comment by dstftw** [comment]
Keep old regexes as fallbacks.

**Comment by YetAnotherGhost** [comment]
Done (as separate commit, since I'm using the web UI). I hope the lines aren't too long.

File: youtube_dl/extractor/eroprofile.py [hunk 1]
```diff
@@ -80,9 +80,10 @@ def _real_extract(self, url):
         video_url = unescapeHTML(self._search_regex(
             r'<source src="([^"]+)', webpage, 'video url'))
         title = self._html_search_regex(
-            r'Title:</th><td>([^<]+)</td>', webpage, 'title')
+            [r'<h1[^>]*>([^<]+)</h1>', r'Title:</th><td>([^<]+)</td>'],
+            webpage, 'title')
         thumbnail = self._search_regex(
-            r'onclick="showVideoPlayer\(\)"><img src="([^"]+)',
+            [r'<div class="playlistItem current">.*<img src="([^"]+)"', r'onclick="showVideoPlayer\(\)"><img src="([^"]+)'],
```
Comment (dstftw) [inline youtube_dl/extractor/eroprofile.py]: Part in the middle should not be greedy. Also use `_parse_html5_media_entries` as main path.

Comment (YetAnotherGhost) [reply youtube_dl/extractor/eroprofile.py]: > Part in the middle should not be greedy.

Done, thanks.

> Also use `_parse_html5_media_entries` as main path.

I'm not sure what you mean… but this change doesn't touch the `video_url` so far?

**Comment by YetAnotherGhost** [comment]
What’s the status on this? I’m done at the moment, as far as I’m concerned – if you’re waiting for anything else from me, let me know what’s still missing before merging this.

**Comment by dstftw** [comment]
I've already pointed out to use `_parse_html5_media_entries` as main path for formats and thumbnail extraction.

**Comment by YetAnotherGhost** [comment]
Yes, and I already said (or tried to say, I guess it wasn’t clear) that a) I wasn’t sure what you meant by that (the “for formats and thumbnail extraction” makes it a bit clearer, thanks), and b) if it means what I think it means, I don’t think it belongs in this pull request, since it’s unrelated beyond affecting the same extractor.

I tried to implement this in YetAnotherGhost/youtube-dl#1 (was supposed to be a pull request against this repository but I didn’t pay attention when creating the pull request; can be moved later). Is it more or less what you were thinking of?

**Pull request closed without merging**